src/operators.jl

```diff
@@ -858,3 +858,191 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+"""
+    linegraph(g)
+
+returns only the line graph of a given graph if returnmap is false
+returns line graph and a dictionary mapping new vertices in line graph to
+old edges in graph if returnmap is true
+"""
+function linegraph(g::AbstractGraph, returnmap::Bool = false)
+    lg = is_directed(g) ? SimpleDiGraph(ne(g)) : SimpleGraph(ne(g))
+    graphE_lineV = Dict(e => i for (i, e) in enumerate(edges(g)))
```

Comment:
Would it be possible to do that with a `Vector` instead?  `Dict`is much slower.

Reply:
I don't think so, every edge in the graph corresponds to a vertex in the line graph, I can't think of a simple approach to do this without using edges.